osf/migrations/0016_backfill_date_retracted.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.9 on 2017-03-14 14:41
+
+"""
+Backfill Retraction.date_retracted with `RETRACTION_APPROVED` log date.
+"""
+
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+from osf.models import Registration, Retraction, Sanction
+
+
+def set_date_retracted(*args):
+    registrations = (
+        Registration.objects.filter(retraction__state=Sanction.APPROVED, retraction__date_retracted=None)
+        .select_related('retraction')
+        .include('registered_from__logs')
+        .include('registered_from__guids')
+    )
+    total = registrations.count()
+    print('Migrating {} retractions.'.format(total))
+
+    for registration in registrations:
```

Comment:
There's probably a way, using [F expressions](https://docs.djangoproject.com/en/1.11/ref/models/expressions/#django.db.models.F), to do this in one query. If this was something that was going to happen more than once I'd suggest giving it a shot to make it work with those. Just a heads up that they exist and can do very fancy things.